packages/core/src/domain/geographic-locations.ts

```diff
@@ -54,3 +54,8 @@ export enum USState {
   // "Guam", "GU"
   // "Puerto Rico", "PR"
 }
+
+export function getStateEnum(state: string): USState | undefined {
+  const upperCaseState = state.toUpperCase();
+  return upperCaseState in USState ? (upperCaseState as USState) : undefined;
```

Comment:
Hmmm I think that gives this error:
`Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'typeof USState'.`
TypeScript doesn't know at compile time that upperCaseState will definitely be a key of the USState enum. 

I used a type guard apporoach instead